cpp/src/io/parquet/experimental/hybrid_scan_impl.cpp

```diff
@@ -225,8 +225,57 @@ hybrid_scan_reader_impl::filter_row_groups_with_dictionary_pages(
 
   select_columns(read_mode::FILTER_COLUMNS, options);
 
-  return _metadata->filter_row_groups_with_dictionary_pages(
-    {}, {}, row_group_indices, {}, {}, {}, options.get_filter().value(), stream);
+  table_metadata metadata;
+  populate_metadata(metadata);
+  auto expr_conv = named_to_reference_converter(options.get_filter(), metadata);
+  CUDF_EXPECTS(expr_conv.get_converted_expr().has_value(),
+               "Columns names in filter expression must be convertible to index references");
+  auto output_dtypes = get_output_types(_output_buffers_template);
+
+  // Collect literals and operators for dictionary page filtering for each input table column
+  auto const [literals, operators] =
+    dictionary_literals_collector{expr_conv.get_converted_expr().value().get(),
+                                  static_cast<cudf::size_type>(output_dtypes.size())}
+      .get_literals_and_operators();
+
+  // Return all row groups if no dictionary page filtering is needed
+  if (literals.empty() or std::all_of(literals.begin(), literals.end(), [](auto& col_literals) {
+        return col_literals.empty();
+      })) {
+    return std::vector<std::vector<size_type>>(row_group_indices.begin(), row_group_indices.end());
+  }
+
+  // Collect schema indices of columns with an (in)equality predicate
```

Comment:
The comments here are a bit confusing; maybe mention that `dictionary_literals_collector` collects (in)equality predicates instead of only mentioning the type of predicate here. Based on the comments I expected this `thrust::copy_if` to be the part that selects (in)equality predicates.